lib/navigation/consts.hpp

```diff
@@ -0,0 +1,47 @@
+#include <cstdint>
+
+namespace hyped::navigation{
+
+  using nav_t = float;
+  static constexpr nav_t kTrackLength       = 100.0; //m
+  static constexpr nav_t kBrakingDistance   = 20.0;  //m
+  static constexpr uint8_t kNumImus         = 4;
+  static constexpr uint8_t kNumEncoders     = 4;
+  static constexpr uint8_t kNumKeyence      = 2;
+  static constexpr nav_t kPi                = 3.14159265359;
+  static constexpr nav_t kWheelCicumference = kPi * 0.1; //m //TODO: check!
+  static constexpr nav_t kStripeDistance    = 10.0; //m
+  
+  //current trajectory struct
+  struct Trajectory {
+  nav_t displacement;
+  nav_t velocity;
+  nav_t acceleration;
+};
+
+//inline trajectory equal to oprator for use in testing
+inline bool operator==(const Trajectory &lhs, const Trajectory &rhs)
+{
+  return lhs.displacement == rhs.displacement && lhs.velocity == rhs.velocity
+         && lhs.acceleration == rhs.acceleration;
+}
+
+
+//inline trajectory not equal to oprator for use in testing
+inline bool operator!=(const Trajectory &lhs, const Trajectory &rhs)
+{
+ return !(lhs == rhs);
+}
+
+//explicitly define zero trajectory
+inline Trajectory zero_trajectory = {0, 0, 0};
+
+
+
+template<typename SensorData>
+class INavigator {
```

Comment:
Do we still need this?

If so, your `Navigator` should implement `INavigator<WheelEncoderData>`, `INavigator<KeyenceData>`, `INavigator<ImuData>`.

The reason this was here to begin with was so that we could compare navigators in a simulated environment. I.e. we can just swap them out and test them against some data. I still believe that would be a decent idea, but it doesn't seem unfixable later on. Just make sure that your interface is intuitive (input the things that you can handle nicely internally and output the things that you produce directly).